frontend/src/pages/Search.js

```diff
@@ -162,239 +155,320 @@ function Search({ location, history }) {
+    filtered = orderBy(
+      filtered,
+      [
+        (listing) => !!listing[key],
+        (listing) => listing[key],
+        (listing) => listing.course_code,
+      ],
+      ['desc', order_asc ? 'asc' : 'desc', 'asc']
+    );
 
-      // Variables to use in search query
-      const search_variables = {
-        search_text: searchText.value,
-        ordering: ordering,
-        seasons: processedSeasons,
-        areas: processedAreas,
-        skills: processedSkills,
-        credits: processedCredits,
-        schools: processedSchools,
-        subjects: processedSubjects,
-        min_rating: include_all_ratings ? null : ratingBounds[0],
-        max_rating: include_all_ratings ? null : ratingBounds[1],
-        min_workload: include_all_workloads ? null : workloadBounds[0],
-        max_workload: include_all_workloads ? null : workloadBounds[1],
-        extra_info: hideCancelled ? 'ACTIVE' : null,
-        fy_sem: hideFirstYearSeminars ? false : null,
-      };
+    console.log('end search/filter');
+    return filtered;
+  }, [
+    required_seasons,
+    coursesLoading,
+    courseLoadError,
+    courseData,
+    searchConfig,
+  ]);
 
-      // Track search
-      posthog.capture('search', {
-        ...search_variables,
-        search_text_clean: search_variables.search_text || '[none]',
-      });
+  const handleSetView = (isList) => {
+    posthog.capture('catalog-view-toggle', { isList });
+    setView(isList);
+  };
 
-      // Execute search query
-      executeSearch({
-        variables: search_variables,
-      });
-    }
-    // [
-    //   executeSearch,
-    //   hideCancelled,
-    //   multiSeasons,
-    //   ratingBounds,
-    //   sort_order,
-    //   select_credits,
-    //   select_schools,
-    //   select_seasons,
-    //   select_skillsareas,
-    //   select_sortby.value,
-    //   select_subjects,
-    //   workloadBounds,
-    // ]
-  );
+  const handleSortOrder = () => {
+    if (sort_order === 'asc') setSortOrder('desc');
+    else setSortOrder('asc');
+  };
 
-  // If the search query was called
-  if (searchCalled) {
-    // If the search query is still loading
-    if (searchLoading) {
-    } else if (searchData) {
-      // Scroll down to catalog when search is complete for mobile view
-      if (isMobile) {
-        scroller.scrollTo('catalog', {
-          smooth: true,
-          duration: 500,
-        });
-      }
-      // Preprocess search data
-      searchData = searchData.search_listing_info.map((x) => {
-        return flatten(x);
-      });
-      searchData = searchData.map((x) => {
-        return preprocess_courses(x);
+  // If the search query is still loading
+  if (coursesLoading) {
+  } else if (searchData) {
+    // Scroll down to catalog when search is complete for mobile view
+    // TODO: this is called too frequently
+    if (isMobile) {
+      scroller.scrollTo('catalog', {
+        smooth: true,
+        duration: 500,
```

Comment:
@maxyuan6717 any idea what to do here now that we have live-updating results?

Reply:
Maybe adding back the submit button on mobile and only filtering/scrolling on submit? So like only update searchConfig on submit

Reply:
It feels like it would complicate data flow a bit - what if we scrolled on first render + anytime they hit "enter" in the search form to submit it, but kept the live updating?

Reply:
ohhh I like that

Reply:
kk will do that